infra/resources/main.tf

```diff
@@ -27,6 +27,11 @@ terraform {
 provider "google" {
 }
 
+provider "google-beta" {
```

Comment:
What requires google beta here?

Reply:
Only the `google-beta` provider for the `google_sql_database_instance` for supports private networking. See [google-sql-database-instance Terraform Doc](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#private-ip-instance)

Reply:
In the example, they pull this provider setting under the resource that actually needs it. Shall we do that here?

Reply:
I want to keep the providers all together `main.tf` because it is easier to manage.